libflux/src/scanner/scanner.rl

```diff
@@ -8,8 +7,22 @@
 
     include WChar "unicode.rl";
 
-    # For comments on newlines tracking, see below.
-    newline = '\n' @{ push(&nls, fpc - data + 1); no_nls++; };
+    action advance_line {
+        // We do this for every newline we find.
+        // This allows us to return correct line/column for each token
+        // back to the caller.
+        (*cur_line)++;
+        *last_newline = fpc + 1;
```

Comment:
I don't like this name 😅 

Could it be something like `last_newline_col`? Just to make it explicit that it refers to the column position